src/Datapackages/BaseDatapackage.php

```diff
@@ -183,20 +183,27 @@ public function save($zip_filename)
     {
         Package::isZipPresent();
         $zip = new ZipArchive();
+
+        $packageCopy = $this->copy();
+
         $base = tempnam(sys_get_temp_dir(), 'datapackage-zip-');
         $files = [
             'datapackage.json' => $base.'datapackage.json',
         ];
         $ri = 0;
-        foreach ($this as $resource) {
+        foreach ($packageCopy->resources() as $resource) {
+            $resourceFiles = [];
             $fileNames = $resource->save($base.'resource-'.$ri);
             foreach ($fileNames as $fileName) {
                 $relname = str_replace($base.'resource-'.$ri, '', $fileName);
                 $files['resource-'.$ri.$relname] = $fileName;
+                $resourceFiles[] = 'resource-'.$ri.$relname;
             }
+            $resource->descriptor()->path = count($resourceFiles) == 1 ? $resourceFiles[0] : $resourceFiles;
```

Comment:
I may need to test this further but let me see if I understand this: `path` property according to the `schema` can either be a string or an array of strings. 


https://github.com/frictionlessdata/datapackage-php/blob/48e73fc9b9878fa7bb624f5e9dd0173962077fad/src/Validators/schemas/data-package.json#L262-L276

If you have only one resource file for a resource you use the the file path/string, if not you use the array, but if fileNames (save) is empty (should not happen really) but  What happens if the resource has also e.g `URIs` for the same resource? 
let's say mytable.csv and https://myrepo.org/data/mytable.csv?

Maybe you should take the original paths from the descriptor first and only replace the original absolute paths that match the files you are iterating over but keep the URIs if any?

Reply:
You're correct, I hadn't considered the case when path is a URI. I'll make sure that's covered in the tests and then work to fix the implementation.

Reply:
So I played around with remote URLs some and it turns out the BaseDatapackage->save() method currently writes _remote_ resources to disk as part of saving (even before my pull request changes). As I understand it, that's not how it's supposed to work, right?

Reply:
Hi. You are right, but i would say (personal opinion) this may be seen as a not so good documented expected behavior/facility that can save some steps for people building Packages. Since `fopen` in PHP can deal with URLs same way as with a local file you get this for free. A datastream level on `::save` you already saw this
https://github.com/frictionlessdata/datapackage-php/blob/48e73fc9b9878fa7bb624f5e9dd0173962077fad/src/DataStreams/DefaultDataStream.php#L46-L51 

Of course it would be nice if one could flag if the local path is going to be the only one to be kept or also the original remote one, 
but does not mean (I guess) that you can not still for a local resource/file add an additional ->path manually to mimic that behavior.

Not sure how the Python and JS versions deal with this. I may want to check with their interfaces to ensure we do this the same way. Probably a different ISSUE/Pull.

Reply:
I agree. probably at the Datastream level we should have flag. Will dig into the other language implementations and see how they do this. Will let you know how this goes. thx